File: Examples/VisualBasic/Rendering-Printing/PrintProgressDialog.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.IO
Imports System.Drawing
Imports Aspose.Words
Imports Aspose.Words.Layout
Imports Aspose.Words.Rendering
Public Class PrintProgressDialog
    Public Shared Sub Run()
        ' ExStart:PrintProgressDialog
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_RenderingAndPrinting()
        ' Load the documents which store the shapes we want to render.
        Dim doc As New Document(dataDir & Convert.ToString("TestFile RenderShape.doc"))
        ' Obtain the settings of the default printer
        Dim settings As New System.Drawing.Printing.PrinterSettings()

        ' The standard print controller comes with no UI
        Dim standardPrintController As System.Drawing.Printing.PrintController = New System.Drawing.Printing.StandardPrintController()

        ' Print the document using the custom print controller
        Dim prntDoc As New AsposeWordsPrintDocument(doc)
        prntDoc.PrinterSettings = settings
        prntDoc.PrintController = standardPrintController
        prntDoc.Print()
        ' ExEnd:PrintProgressDialog
    End Sub
End Class

File: Examples/CSharp/Rendering-Printing/PrintProgressDialog.cs
using System;
using System.IO;
using System.Drawing;
using Aspose.Words;
using Aspose.Words.Layout;
using Aspose.Words.Rendering;
using Aspose.Words.Saving;
using Aspose.Words.Drawing;
using System.Drawing.Imaging;
using Aspose.Words.Tables;

namespace Aspose.Words.Examples.CSharp.Rendering_and_Printing
{
    class PrintProgressDialog
    {
        public static void Run()
        {
            //ExStart:PrintProgressDialog
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_RenderingAndPrinting(); 
            // Load the documents which store the shapes we want to render.
            Document doc = new Document(dataDir + "TestFile RenderShape.doc");
            // Obtain the settings of the default printer
            System.Drawing.Printing.PrinterSettings settings = new System.Drawing.Printing.PrinterSettings();

            // The standard print controller comes with no UI
            System.Drawing.Printing.PrintController standardPrintController = new System.Drawing.Printing.StandardPrintController();

            // Print the document using the custom print controller
            AsposeWordsPrintDocument prntDoc = new AsposeWordsPrintDocument(doc);
            prntDoc.PrinterSettings = settings;
            prntDoc.PrintController = standardPrintController;
            prntDoc.Print();            
            //ExEnd:PrintProgressDialog
        }
        
    }
}
